torch/csrc/autograd/variable.cpp

```diff
@@ -363,6 +363,24 @@ void clear_hooks(const at::TensorBase& self) {
   materialize_autograd_meta(self)->hooks_.clear();
 }
 
+void add_post_acc_grad_hook(
+    const at::TensorBase& self,
+    std::unique_ptr<PostAccumulateGradHook> hook) {
+  AutogradMeta* meta = materialize_autograd_meta(self);
```

Comment:
Assert here that post_acc_grad_hooks_ is empty.

Reply:
wait...why?

Reply:
Don't we want multiple hooks to be able to get added?

Reply:
cuz hook here...isn't a dict right?

Reply:
Ah....it is weird
okay i'm adding some comments